src/modules/reducers/legacyBalances.js

```diff
@@ -0,0 +1,156 @@
+export const LEGACY_BALANCE_CONVERSION_SUCCESS = "legacyBalances/LEGACY_BALANCE_CONVERSION_SUCCESS";
+
+const initialState = {
+    refreshError: null,
+    error: null,
+    isLoading: false,
+    contractBalances: null,
+    result: null,
+    request: null
+};
+
+export default (state = initialState, action) => {
+    switch (action.type) {
+        case LEGACY_BALANCE_REFRESH_REQUESTED:
+            return {
+                ...state,
+                isLoading: true,
+                refreshError: null
+            };
+
+        case LEGACY_BALANCE_REFRESH_SUCCESS:
+            return {
+                ...state,
+                contractBalances: action.result,
+                isLoading: false,
+                refreshError: null
+            };
+
+        case LEGACY_BALANCE_REFRESH_ERROR:
+            return {
+                ...state,
+                isLoading: false,
+                refreshError: action.error
+            };
+
+        case LEGACY_BALANCE_CONVERSION_REQUESTED:
+        case LEGACY_BALANCE_DISMISS_REQUESTED:
+            return {
+                ...state,
+                request: action.request
+            };
+
+        case LEGACY_BALANCE_CONVERSION_SUCCESS:
+            return {
+                ...state,
+                result: action.result,
+                request: null
+            };
+
+        case LEGACY_BALANCE_CONVERSION_ERROR:
+        case LEGACY_BALANCE_DISMISS_ERROR:
+            return {
+                ...state,
+                isLoading: false,
+                error: action.error,
+                request: null
+            };
+
+        case LEGACY_BALANCE_DISMISS_SUCCESS:
+            return {
+                ...state,
+                contractBalances: action.result,
+                request: null
+            };
+
+        default:
+            return state;
+    }
+};
+
+export const refreshLegacyBalances = () => {
+    return async dispatch => {
+        dispatch({
+            type: LEGACY_BALANCE_REFRESH_REQUESTED
+        });
+        try {
+            const contractBalances = await fetchLegacyBalances();
+            return dispatch({
+                type: LEGACY_BALANCE_REFRESH_SUCCESS,
+                result: contractBalances
+            });
+        } catch (error) {
+            if (process.env.NODE_ENV !== "production") {
+                return Promise.reject(error);
+            }
+            return dispatch({
+                type: LEGACY_BALANCE_REFRESH_ERROR,
+                error: error
+            });
+        }
+    };
+};
+
+export function dismissLegacyBalance(legacyTokenAddress) {
+    return async dispatch => {
+        dispatch({
+            type: LEGACY_BALANCE_DISMISS_REQUESTED,
+            request: { legacyTokenAddress }
+        });
+        try {
+            const contractBalances = [...store.getState().legacyBalances.contractBalances];
+            // TODO: shall we use tokenAddress as key instead of using an array?
```

Comment:
you mean like here? https://github.com/Augmint/augmint-web/pull/149/files#diff-fdea7c4c7ec1e3df4eaf07c4d9247127R21